Dockerfile.devel

```diff
@@ -0,0 +1,97 @@
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+# STAGE 1
+FROM       artifactory.algol60.net/docker.io/library/alpine:3.17 as builder
+RUN        apk add --update --no-cache \
+             cmake=3.24.3-r0 \
+             g++=12.2.1_git20220924-r4 \
+             gcc=12.2.1_git20220924-r4 \
+             git=2.38.3-r1 \
+             libffi-dev=3.4.4-r0 \
+             make=4.3-r1 \
+             musl-dev=1.2.3-r4 \
+             openssl=3.0.7-r2 \
+             py3-pip=22.3.1-r1 \
+             python3=3.10.9-r1 \
+             python3-dev=3.10.9-r1
+# make canu group, create (system) canu user, add to canu group, make home folder, and set login shell, no password
+RUN        addgroup -S canu 
+RUN        adduser \
+             -S canu \
+             -G canu \
+             -h /home/canu \
+             -s /bin/bash \
+             -D
+USER       canu
+RUN        mkdir -p /home/canu/app/wheels
+WORKDIR    /home/canu/app
+# copy the source code into the container
+COPY       --chown=canu:canu . .
+# Install the build dependencies as wheels
+ENV        PYTHONDONTWRITEBYTECODE 1
+ENV        PYTHONUNBUFFERED 1
+# Specific to canu-in-a-container, wheel is used to reduce the final image size
+# by building the dependencies as wheels and copying them into stage2, eliminating
+# the need to install the build dependencies in stage2
+RUN        python -m pip install --user -U wheel
+RUN        python -m pip --no-cache-dir wheel . '.[build-system]' --wheel-dir wheels/
+
+# STAGE 2
+FROM       artifactory.algol60.net/docker.io/library/alpine:3.17 as prod
+RUN        apk add --update --no-cache \
+              git=2.38.3-r1 \
+              py3-pip=22.3.1-r1 \
+              python3=3.10.9-r1
+# add the canu user and group to the prod container
+RUN        addgroup -S canu 
+RUN        adduser \
+             -S canu \
+             -G canu \
+             -h /home/canu \
+             -s /bin/bash \
+             -D
+USER       canu
+RUN        mkdir -p /home/canu/app
+RUN        mkdir -p /home/canu/mounted
+WORKDIR    /home/canu/app
+# copy the wheels from the builder container
+COPY       --chown=canu:canu --from=builder /home/canu/app/wheels /home/canu/app/wheels
+COPY       --chown=canu:canu . .
+# install using the wheels compiled from stage 1
+ENV        PATH "/home/canu/.local/bin:$PATH"
+ENV        PYTHONDONTWRITEBYTECODE 1
+ENV        PYTHONUNBUFFERED 1
+# not all wheels are possible on alpine: https://github.com/pypa/manylinux/issues/37
+RUN        python -m pip --no-cache-dir install --user --find-links=wheels/ .
+# add a script that installs in editable mode for ad-hoc debugging
+RUN        install -m 0755 -o canu -g canu /home/canu/app/canu-debug /home/canu/.local/bin/canu-debug
+WORKDIR    /home/canu
+ENV        PS1 "canu \w$ "
+# env vars for canu-inventory
+ENV        CANU_NET "HMN"
```

Comment:
You can define multiple environment variables with a single `ENV`, it'll shrink the Docker image layer count down.

```Dockerfile
ENV key=value \
    key=value \
    key=value
```
This'll make the ENV setting a single layer for cleaner build output.

Reply:
Fixed